src/sentry/integrations/manager.py

```diff
@@ -35,17 +35,17 @@ def exists(self, key: str) -> bool:
         return key in self.__values
 
     def register(self, cls: type[IntegrationProvider]) -> None:
-        self.__values[cls.key] = cls
+        self.__values[str(cls.key)] = cls
```

Comment:
hmmm the `str(...)` calls are suspicious here -- is this defined as `str | None` somewhere? should it be `: str` and that'd prevent the `str` calls?

Reply:
i think it's because `cls` is `type[IntegrationProvider]` rather than just `IntegrationProvider`? if i don't wrap it with `str` mypy thinks `cls.key` is `Callable[[PipelineProvider], str]`

Reply:
ah, yeah it isn't understanding the `abstractmethod` + `property`

maybe this will fix it?
```
diff --git a/src/sentry/pipeline/provider.py b/src/sentry/pipeline/provider.py
index b08693aef9..3f9df77320 100644
--- a/src/sentry/pipeline/provider.py
+++ b/src/sentry/pipeline/provider.py
@@ -18,13 +18,12 @@ class PipelineProvider(abc.ABC):
     def __init__(self) -> None:
         self.config: dict[str, Any] = {}
 
-    @property
-    @abc.abstractmethod
-    def key(self) -> str:
-        """
-        A unique identifier (e.g. 'slack'). Used to lookup sibling classes and
-        the `key` used when creating Integration objects.
-        """
+    # abstract
+    """
+    A unique identifier (e.g. 'slack'). Used to lookup sibling classes and
+    the `key` used when creating Integration objects.
+    """
+    key: str
 
     @property
     @abc.abstractmethod
 ```

Reply:
it worked 🤯